lib/msal-node/docs/caching.md

```diff
@@ -0,0 +1,109 @@
+# Token caching in MSAL Node
+
+When MSAL Node acquires a token, it caches it in memory for future usage. MSAL Node manages the token lifetime and refreshing for you. APIs like `acquireTokenSilent()` retrieves the access tokens from the cache for a given account:
+
+```javascript
+const msal = require('@azure/msal-node');
+
+// Create msal application object
+const cca = new msal.ConfidentialClientApplication({
+    auth: {
+        clientId: "Enter_the_Application_Id_Here",
+        authority: "https://login.microsoftonline.com/Enter_the_Tenant_Info_Here",
+        clientSecret: "Enter_the_Client_Secret_Here"
+    }
+});
+
+const someUserHomeAccountId = "Enter_User_Home_Account_Id";
+
+const msalTokenCache = cca.getTokenCache();
+const account = await msalTokenCache.getAccountByHomeId(someUserHomeAccountId)
+
+const silentTokenRequest = {
+    account: account,
+    scopes: ["user.read"],
+};
+
+cca.acquireTokenSilent(silentTokenRequest).then((response) => {
+    // do something with response
+}).catch((error) => {
+    // catch and handle errors
+});
+```
+
+In-memory token cache is not suitable for production. In production, you should serialize and persist the token cache. Depending on the type of application, you have several alternatives:
```

Comment:
Redis was in-memory, no?  Are we really recommending folks persist tokens in a DB?  I wouldn't have suggested that.  Maybe we should state that a single, per-server cache instance isn't suitable for a cloud environment with many servers?  Maybe we should suggest they store cached tokens in an ephemeral, in-memory cache like Redis that can be distributed across all their servers.  @sameerag and @samuelkubai, did you have other thoughts on this?

Reply:
Technically yes : ) I've added a qualifier.

It seems using a DB for caching is common ([see](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-5.0#distributed-memory-cache)). It also seems common to use Redis-like caches and DBs in tandem, as different layers of persistence.

Reply:
Well ... Couple of points:
* We should call out in-memory as not scalable rather than not suitable to production. For public clients, there can be users who do not want to ever persist their cache
* DB discussion depends on who we are talking to. I would highlight encryption or the necessity of it in all these use cases.

Reply:
got it, will update

Reply:
@sameerag revised